posts/introduction-to-grpc.md

```diff
@@ -0,0 +1,225 @@
+file systems and sets up a bridge network where they will communicate. In an actual deployed environment, we will
+likely have to use some sort of DNS to reference the host.
+
+
+All of code samples from this article are snippets of our previously linked project which can be used as a
+reference. You can find on the application available on repository, shared on this article, some additional configurations
+related to the familiar spring patterns. Among other things, we have a `@Configuration` class which is used to configure
+our gRPC server to be running inside our spring application. In this configuratioon, `DerivativeCalculatorService`
+houses our previously shown derivative logic.
+
+```java
+@Configuration
+public class GrpcServerConfiguration {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GrpcServerConfiguration.class);
+
+    private int port;
+    private Server server;
+    private DerivativeCalculatorService derivativeCalculatorService;
+
+    public GrpcServerConfiguration(ApplicationProperties applicationProperties, DerivativeCalculatorService derivativeCalculatorService) {
+        this.port = applicationProperties.getGrpc().getPort();
+        this.derivativeCalculatorService = derivativeCalculatorService;
+    }
+
+    @Bean
+    public Server grpcServer() throws IOException {
+        LOGGER.info("Starting GRPC Server");
+        this.server = ServerBuilder.forPort(this.port).addService((BindableService) derivativeCalculatorService).build();
+        this.server.start();
+        return this.server;
+    }
+}
+```
+
+Now we can finally start up our spring applications and see a message sent from our gRPC client to the gRPC server.
+First we will want to build our two applications with maven, using the common `mvn clean install` (this will compile the
+protobuffer), then we can start our two images with the command `docker-compose -f docker-compose.yml up —build`.
+Each project in our GitHub has its own docker compose file, one in server/docker-compose, and one in client/docker-compose.
+With both up and running, we can send requests to our client (also a spring boot app), via a simple rest endpoint.
+It takes a polynomial in the form of a string (which is cumbersome as we said before), converts to our protobuf format
+and sends the polynomial to the server. Assuming these are both up and running, we simply run the following command
+inside our terminal:
+
+```
+curl --location --request POST 'localhost:8081/example/calculate-derivative' \
+--header 'Content-Type: application/json' \
+--data-raw '{"polynomial": "1x^4 + 2x^2 + 1x^1 + 1x^0"}'
+```
+
+Feel free to play around with the example and try other simple polynomials. Some caveats when using this, be aware that
+some requirements exist, for one, our polynomials we send include all exponents and coefficients (even 1 and 0), and
+that’s why we see `1x^1 + 1x^0` instead of `x + 1`, this is simply due to our simple parsing of the string polynomial.
```

Comment:
```suggestion
Feel free to play around with the example and try other simple polynomials. Some caveats when using this, be aware that the polynomials we send to the server must include all exponents and coefficients (even 1 and 0), and
that’s why we see `1x^1 + 1x^0` instead of `x + 1`. This is simply due to our simple parsing of the string polynomial.
```